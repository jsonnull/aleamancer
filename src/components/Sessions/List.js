// @flow
import React from 'react'
import styled from 'styled-components'
import Item from './ListItem'
import type { SessionInfo } from 'types'

const EmptyList = styled.div`margin: 2rem 0;`

const List = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin-left: -1rem;
  margin-right: -1rem;
`

type Props = {
  sessions: Array<SessionInfo>,
  setSession: Function
}

const FullList = (props: Props) => {
  const sessions = props.sessions

  if (sessions.length === 0) {
    return (
      <EmptyList>
        {
          "Yikes, looks like you're not a member of any games. Want to start one?"
        }
      </EmptyList>
    )
  }

  return (
    <List>
      {sessions.map(session => (
        <Item
          key={session.id}
          isCurrent={false}
          session={session}
          setSession={props.setSession}
        />
      ))}
    </List>
  )
}

export default FullList
