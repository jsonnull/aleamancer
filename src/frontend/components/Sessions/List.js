// @flow
import React from 'react'
import styled from 'styled-components'
import type { SessionInfo } from 'common/types'
import Item from './ListItem'

const EmptyList = styled.div`
  margin: 2rem 0;
`

const List = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin-left: -1rem;
  margin-right: -1rem;
`

const Loading = () => <div>Loading...</div>

type Props = {
  data: { loading: boolean, error?: string, variables: Object },
  setSession: Function
}

const FullList = (props: Props) => {
  const { loading, error, currentUser } = props.data

  if (loading) {
    return <Loading />
  }

  if (error) {
    throw error
  }

  const sessions = currentUser.games

  if (sessions.length === 0) {
    return (
      <EmptyList>
        {
          "Yikes, looks like you're not a member of any games. Want to start one?"
        }
      </EmptyList>
    )
  }

  return (
    <List>
      {sessions.map(session => (
        <Item
          key={session.id}
          name={session.name}
          setSession={() => props.setSession(session.id)}
        />
      ))}
    </List>
  )
}

export default FullList
