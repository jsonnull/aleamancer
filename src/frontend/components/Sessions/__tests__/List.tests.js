// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import List from '../List.js'

describe('Sessions List component', () => {
  const currentUserWithNoSessions = {
    games: []
  }
  it('renders correctly with no sessions', () => {
    const tree = renderer
      .create(
        <List
          loading={false}
          currentUser={currentUserWithNoSessions}
          setSession={() => {}}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  const currentUser = {
    games: [
      {
        id: 'id',
        name: 'testName'
      }
    ]
  }
  it('renders correctly with sessions', () => {
    const tree = renderer
      .create(
        <List loading={false} currentUser={currentUser} setSession={() => {}} />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
