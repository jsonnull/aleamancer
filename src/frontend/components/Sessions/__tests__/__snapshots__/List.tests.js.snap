// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sessions List component renders correctly with no sessions 1`] = `
.c0 {
  margin: 2rem 0;
}

<div
  className="c0"
>
  Yikes, looks like you're not a member of any games. Want to start one?
</div>
`;

exports[`Sessions List component renders correctly with sessions 1`] = `
.c2 {
  font-family: Lora;
  font-size: 1.8rem;
}

.c1 {
  padding: 1rem;
  border-radius: 5px;
  margin: 0 1rem 2rem;
  -webkit-flex: 1 0 25%;
  -ms-flex: 1 0 25%;
  flex: 1 0 25%;
  max-width: 25%;
  min-height: 100px;
  cursor: pointer;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-left: -1rem;
  margin-right: -1rem;
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
  >
    <div>
      <div
        className="c2"
      >
        testName
      </div>
    </div>
  </div>
</div>
`;
